utils/mapitest/modules/module_zentyal.c

```diff
@@ -587,3 +588,164 @@ _PUBLIC_ bool mapitest_zentyal_6723(struct mapitest *mt)
 
 	return test_result;
 }
+
+/**
+    \details Test #8174 ModifyRecipients and try to build RecipientRow
+    with no additional properties.
+
+    It is identified as CR-1572 (https://projects.zentyal.com/issues/3681)
+
+    This function does:
+    -# Open Drafts folder
+    -# Create a message
+    -# Set a recipient with some additional properties such as PidTagObjectType
+    -# Set a recipient with no additional properties
+    -# Call ModifyRecipients with these two recipients
+    -# Delete the message
+
+    \param mt pointer to the top level mapitest structure
+
+    \return true on success, otherwise false
+*/
+_PUBLIC_ bool mapitest_zentyal_8174(struct mapitest *mt)
+{
+	bool				ret = true;
+	char				**username = NULL;
+	enum MAPISTATUS			retval;
+	mapi_object_t			obj_store;
+	mapi_object_t			obj_folder;
+	mapi_object_t			obj_message;
+	mapi_id_t			id_folder;
+	mapi_id_t			id_msgs[1];
+	struct PropertyValue_r		value;
+	struct PropertyRowSet_r		*row_set = NULL;
+	struct PropertyTagArray_r	*flaglist = NULL;
+	struct SPropTagArray		*s_prop_tag_array = NULL;
+	struct SRowSet			*s_row_set = NULL;
+	uint32_t			prop_i;
+
+	/* Step 1. Logon */
+	mapi_object_init(&obj_store);
+	mapi_object_init(&obj_folder);
+	mapi_object_init(&obj_message);
+
+	retval = OpenMsgStore(mt->session, &obj_store);
+	mapitest_print_retval(mt, "OpenMsgStore");
+	if (GetLastError() != MAPI_E_SUCCESS) {
+		ret = false;
+		goto cleanup;
+	}
+
+	/* Step 2. Open Drafts folder */
+	retval = GetDefaultFolder(&obj_store, &id_folder, olFolderDrafts);
+	mapitest_print_retval(mt, "GetDefaultFolder");
+	if (GetLastError() != MAPI_E_SUCCESS) {
+		ret = false;
+		goto cleanup;
+	}
+
+	retval = OpenFolder(&obj_store, id_folder, &obj_folder);
+	mapitest_print_retval(mt, "OpenFolder");
+	if (GetLastError() != MAPI_E_SUCCESS) {
+		ret = false;
+		goto cleanup;
+	}
+
+	/* Step 3. Create the message */
+	retval = CreateMessage(&obj_folder, &obj_message);
+	mapitest_print_retval(mt, "CreateMessage");
+	if (GetLastError() != MAPI_E_SUCCESS) {
+		ret = false;
+		goto cleanup;
+	}
+
+	/* Step 4. Set a valid recipient */
+	s_prop_tag_array = set_SPropTagArray(mt->mem_ctx, 0xA,
+					     PR_ADDRTYPE_UNICODE,
+					     PR_DISPLAY_NAME_UNICODE,
+					     PR_7BIT_DISPLAY_NAME_UNICODE,
+					     PR_EMAIL_ADDRESS_UNICODE,
+					     PR_ENTRYID,
+					     PR_SEND_RICH_INFO,
+					     PR_TRANSMITTABLE_DISPLAY_NAME_UNICODE,
+					     /* These ones are not required by RecipientRow field */
+					     PR_SMTP_ADDRESS_UNICODE,
+					     PR_OBJECT_TYPE,
+					     PR_DISPLAY_TYPE
+		);
+
+	username = talloc_array(mt->mem_ctx, char *, 2);
```

Comment:
not important but with talloc_zero_array the `username[1] = NULL` will be for free. I always like free things